### PR TITLE
fix(FEC-13915): update preact and dependencies

### PR DESCRIPTION
**Pull request by semarche-kaltura**
### Description of the Changes

UPD playkit-js-download plugin to use preact@10.4.6

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/download-plugin-manager.tsx [hunk 1]
```diff
@@ -36,11 +43,13 @@ class DownloadPluginManager extends KalturaPlayer.core.FakeEventTarget {
 
   downloadFile(downloadUrl: string, fileName: string) {
     try {
+      // @ts-expect-error - Property 'config' is protected and only accessible within class 'BasePlugin' and its subclasses.
```
[the remark this answers is not in the document]
Comment (semarche-kaltura) [reply src/download-plugin-manager.tsx:46]: out TS definitions (that makes automatically by API-extractor) are not perfect, I believe that's the root of the TS error

File: webpack.config.js [hunk 2]
```diff
@@ -10,51 +11,70 @@ const plugins = [
...
+              injectType: 'singletonStyleTag'
             }
           },
           {
             loader: 'css-loader',
             options: {
+              esModule: true,
               modules: {
-                localIdentName: '[name]__[local]___[hash:base64:5]',
+                localIdentName: `${CSS_MODULE_PREFIX}-ap-[local]`,
```
Comment (SivanA-Kaltura) [inline webpack.config.js:71]: why is this change needed ?

File: src/download-plugin-manager.tsx [hunk 3]
```diff
@@ -1,32 +1,39 @@
...
-class DownloadPluginManager extends KalturaPlayer.core.FakeEventTarget {
+class DownloadPluginManager extends core.FakeEventTarget {
   private _showOverlay = false;
   private downloadService: any;
   private downloadMetadata: DownloadMetadata = null;
   private playOnClose = false;
 
   constructor(public downloadPlugin: Download) {
     super();
+    // @ts-expect-error - Property 'player' is protected and only accessible within class 'BasePlugin' and its subclasses.
```
Comment (SivanA-Kaltura) [inline src/download-plugin-manager.tsx:22]: you could also pass player, config and logger to the constructor. what i don't like about ts-ignore (and i'm not sure if ts-expect-error behaves the same way) is that it will ignore ANY error on that line, leading to possible bugs in the future

File: src/download.tsx [hunk 4]
```diff
@@ -104,14 +106,14 @@ class Download extends KalturaPlayer.core.BasePlugin {
...
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        //@ts-expect-error - TS2353: Object literal may only specify known properties, and ariaLabel does not exist in type IconDto
         ariaLabel: (<Text id="download.download">Download</Text>) as never,
         displayName: 'Download',
         order: 40,
         svgIcon: {
           path: DOWNLOAD
         },
         onClick: this._handleClick as any,
+        // @ts-expect-error - Type '() => h.JSX.Element' is not assignable to type 'ComponentClass<Record<string, never>, {}> | FunctionalComponent<Record<string, never>>'
```
Comment (SivanA-Kaltura) [inline src/download.tsx:116]: can this be resolved with a cast (as any or similar) ?

File: package.json [hunk 5]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "@playkit-js/playkit-js-downloads",
   "version": "1.3.8",
-  "main": "dist/playkit-downloads.js",
+  "main": "playkit-js-downloads",
```
Comment (SivanA-Kaltura) [inline package.json:4]: why ? the file which webpack outputs is named playkit-downloads, no ?

also the filename from webpack config will affect the plugin name, and this plugin is already in production, so changing it will break backward compatibility

File: package.json [hunk 6]
```diff
@@ -26,33 +26,45 @@
...
+    "@playkit-js/kaltura-player-js": "3.17.24",
+    "@playkit-js/ui-managers": "1.6.0",
     "@typescript-eslint/eslint-plugin": "^5.58.0",
     "@typescript-eslint/parser": "^5.58.0",
+    "babel-loader": "^9.1.3",
     "conventional-github-releaser": "3.1.3",
-    "css-loader": "^6.7.3",
-    "cypress": "^13.6.4",
+    "css-loader": "^6.9.0",
+    "cypress": "^13.7.3",
```
Comment (SivanA-Kaltura) [inline package.json:45]: better use explicit cypress 13.13.1 to avoid the firefox timeout issue in case the caret gets resolved to another version

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

**Review by SivanA-Kaltura — DISMISSED** [review]

**Review by SivanA-Kaltura — APPROVED** [review]

**Review by lianbenjamin — APPROVED** [review]
